components/ipfs/ipfs_utils.cc

```diff
@@ -29,10 +29,37 @@ GURL AppendLocalPort(const std::string& port) {
   return gateway.ReplaceComponents(replacements);
 }
 
+const int kCIDv0QLength = 46;
+const int kCIDv1zLength = 49;
+const int kCIDv1bLength = 59;
```

Comment:
Matching length won't work, because CID can represent different hash functions (default is sha-256), or even have Short data inlined via "identity" hash (like https://cid.ipfs.io/#bafkqaaa)

I believe `bafkqaaa` is the shortest possible CID, you can set that as a minimum length, but direct length check should be removed.